src/main/java/com/google/cloud/resourcemanager/OperationInfo.java

```diff
@@ -0,0 +1,276 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.resourcemanager;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.api.services.cloudresourcemanager.model.Operation;
+import com.google.common.base.MoreObjects;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+/** Google Resource Manager Operation metadata. */
+public class OperationInfo {
+
+  private final Boolean done;
+  private final Status error;
+  private final Map<String, Object> metadata;
```

Comment:
The types here need to be semantically documented. What's the string? What's the object? 

Also, could we have a less generic type? That is, a class rather than a Map.

Reply:
Do you mean, we should have a wrapper class for these Map field, Something like this:

```
class KeyValue {
    private final Map<K,V> keyValue = new HashMap<>();

    public KeyValue(Map<K, V> keyValues) {
      this.keyValue.putAll(keyValues);
    }

    Map<K, V> getKeyValue() {
      return keyValue;
    }
  }
```

Reply:
No, that just reinvents Map which doesn't help. You should use a class that has meaning in the domain and is not a general data structure. I'm not familiar with the domain so I'm not sure what those would be here, but for example if you were connecting names to annual income, instead of Map<String, Integer> you could use something like List<Employee> where

```
class Employee {
  String name;
  Money salary;
}
```

Reply:
These fields come as a batch and we set those in protobuf to send for RPC call. Also, the protobuf fields expect `Map<String, Object>` so after creating a custom class, we will have to use Map as we need to convert collected data for protobuf.

Does this sound better?